src/v/crypto/include/crypto/crypto.h

```diff
@@ -227,4 +227,32 @@ void verify_update(sign_context& ctx, bytes_view msg);
  * @note This function will assert if @p ctx has already been finalized
  */
 bool verify_final(sign_context& ctx, bytes_view signature);
+
+///////////////////////////////////////////////////////////////////////////////
+/// DRBG operation
+///////////////////////////////////////////////////////////////////////////////
+
+/**
+ * Used to generate some random data and stick it into a provided buffer
+ *
+ * @param buf The buffer to place random data into
+ * @param is_private Whether or not to use the 'private' DRBG
+ * @note For private vs public, please see:
+ * https://www.openssl.org/docs/man3.0/man3/RAND_priv_bytes.html
+ * @return bytes_span<> The buffer
+ * @throws crypto_exception On internal error
+ */
+bytes_span<> generate_random(bytes_span<> buf, bool is_private = false);
+
+/**
+ * Generates the requested number of random bytes
+ *
+ * @param len The amount of data to generate
+ * @param is_private Whether or not to use the 'private' DRBG
+ * @note For private vs public, please see:
+ * https://www.openssl.org/docs/man3.0/man3/RAND_priv_bytes.html
+ * @return std::vector<uint8_t> The buffer of random data
+ * @throws crypto_exception On internal error
+ */
+std::vector<uint8_t> generate_random(size_t len, bool is_private = false);
```

Comment:
Maybe `iobuf` is a more useful return type?

Reply:
Because of how you call OpenSSL to get random data, the buffer would have to be contiguous.

Reply:
or bytes?